pluma/pluma-app.c

```diff
@@ -653,8 +654,24 @@ is_in_viewport (PlumaWindow  *window,
 	x += vp_x;
 	y += vp_y;
 
-	sc_width = WidthOfScreen (gdk_x11_screen_get_xscreen (screen));
-	sc_height = HeightOfScreen (gdk_x11_screen_get_xscreen (screen));
+	#ifdef GDK_WINDOWING_WAYLAND
+	if (GDK_IS_WAYLAND_DISPLAY (gdk_display_get_default ()))
+	{
+		GdkMonitor *monitor = gdk_display_get_monitor(gdk_display_get_default (), 0);
+		sc_width = gdk_monitor_get_width_mm(monitor);
+		sc_height = gdk_monitor_get_height_mm(monitor);
```

Comment:
As another reviewer pointed out, getting the size in mm is probably wrong. Upon further inspection though, I suspect this whole block will be useless on Wayland. GTK has no way of knowing the window position on Wayland, so what this is trying to do can't be meaningfully implemented. You should probably just return true if the workspaces match.